packages/terra-form-select/src/combobox/Menu.jsx

```diff
@@ -356,14 +356,16 @@ class Menu extends React.Component {
     }
 
     if (element) {
+      const index = active.getAttribute('index');
+      const totalOptions = optGroupElement ? optGroupElement.props.children.length : document.querySelectorAll('[data-terra-select-option]').length;
       if (element.props.display === '' && element.props.value === '') {
         // Used for case where users selects clear option and opens
         // dropdown again and navigates to clear option
         visuallyHiddenComponent.current.innerText = clearSelectTxt;
       } else if (this.isActiveSelected()) {
-        visuallyHiddenComponent.current.innerText = intl.formatMessage({ id: 'Terra.form.select.selectedText' }, { text: displayText });
+        visuallyHiddenComponent.current.innerText = `${intl.formatMessage({ id: 'Terra.form.select.selectedText' }, { text: displayText })}${index}of${totalOptions}`;
```

Comment:
This new addition to the sentence will probably need translated

Reply:
We can use the existing translations we already have in [terra-paginator](https://github.com/cerner/terra-core/tree/main/packages/terra-paginator/translations) since the format and usage is the same, leaving out the word for "Page":
https://github.com/cerner/terra-core/blob/2511a56ab492730cc93fafd2c029e81d7115ec17/packages/terra-paginator/translations/en-US.json#L7

And since we are concatenating it with the existing string/s, we'll also want to add space and punctuation (parentheses work for all the Terra supported languages) so that the final text is read back more appropriately : e.g. "Group Shades of Green, Forest Green (1 of 5),"   

\
\
Sample i18n with parens:
```
  "Terra.form.select.displayPositionInSetSize": "({indexNumber} of {totalNumber})",
```

\
\
The concatenation can keep the space which might get lost in the translation:
```
visuallyHiddenComponent.current.innerText = `${intl.formatMessage({ id: 'Terra.form.select.selectedText' }, { text: displayText })} ${intl.formatMessage({ id: 'Terra.form.select.displayPositionInSetSize' }, { indexNumber: index, totalNumber: totalOptions })},`;
```

and

```
visuallyHiddenComponent.current.innerText = `${displayText} ${intl.formatMessage({ id: 'Terra.form.select.displayPositionInSetSize' }, { indexNumber: index, totalNumber: totalOptions })},`;
```